src/txSerializers.ts

```diff
@@ -77,36 +80,47 @@ const serializeTxCertificate = (certificate: Certificate) => {
     }
 }
 
+const serializeCollateral = (collateral: Collateral) =>
+    [collateral.transactionId, collateral.index]
+
 export const serializeTxBody = (txBody: TransactionBody) => new Map(([
-    [0, txBody.inputs.map(serializeTxInput)],
-    [1, txBody.outputs.map(serializeTxOutput)],
-    [2, identity(txBody.fee)],
-    [3, identity(txBody.ttl)],
-    [4, txBody.certificates?.map(serializeTxCertificate)],
-    [5, txBody.withdrawals && serializeWithdrawals(txBody.withdrawals)],
-    [6, identity(txBody.update)],
-    [7, identity(txBody.metadataHash)],
-    [8, identity(txBody.validityIntervalStart)],
-    [9, txBody.mint && serializeMultiasset(txBody.mint)],
+    [TransactionBodyKeys.INPUTS, txBody.inputs.map(serializeTxInput)],
+    [TransactionBodyKeys.OUTPUTS, txBody.outputs.map(serializeTxOutput)],
+    [TransactionBodyKeys.FEE, identity(txBody.fee)],
+    [TransactionBodyKeys.TTL, identity(txBody.ttl)],
+    [TransactionBodyKeys.CERTIFICATES, txBody.certificates?.map(serializeTxCertificate)],
+    [TransactionBodyKeys.WITHDRAWALS, txBody.withdrawals && serializeWithdrawals(txBody.withdrawals)],
+    [TransactionBodyKeys.UPDATE, identity(txBody.update)],
+    [TransactionBodyKeys.METADATA_HASH, identity(txBody.metadataHash)],
+    [TransactionBodyKeys.VALIDITY_INTERVAL_START, identity(txBody.validityIntervalStart)],
+    [TransactionBodyKeys.MINT, txBody.mint && serializeMultiasset(txBody.mint)],
+    [TransactionBodyKeys.SCRIPT_DATA_HASH, txBody.scriptDataHash],
+    [TransactionBodyKeys.COLLATERAL_INPUTS, txBody.collaterals?.map(serializeCollateral)],
+    [TransactionBodyKeys.REQUIRED_SIGNERS, txBody.requiredSigners],
+    [TransactionBodyKeys.NETWORK_ID, txBody.networkId],
 ]).filter(([_, value]) => value !== undefined) as [number, unknown][])
 
-export const serializeTx = (tx: Transaction) => [
-    serializeTxBody(tx.body),
-    tx.witnessSet,
-    tx.auxiliaryData,
-]
+export const serializeTx = (tx: Transaction) => {
```

Comment:
Don't we want to mention the return type here?
I am unsure about the implications, but this change is breaking, i.e. the value returned from serializeTx is specific to some era or something. Do we document that? Or is `serializeTx` not part of the public API? Or could it perhaps be detected whether the given input tx is compatible with this output format?...

Reply:
OK, that seems reasonable.